### DISCUSSION TITLE
Absolute and relative links

### DISCUSSION THREAD

**Discussion by EdoFro** [EdoFro/Freeplane_Map-Drive-Inator#13] [General]
Hi,
I will add to MDI the capability to work with relative links.

Here is what I have thought untill now:

- I can't imagine a use case where a user wants to have absolute and relative links in the same MDI branch
- MDI edits nodes links when moving files, when adding files to the mind map and when creating files. In those cases it has to know if the link should be absolute or relative.
- I will add an attribute to the baseFolder node where the user can define "absolute or relative" preference
- The default value for that attribute should be taken from Freeplane preferences
- if the user paste some nodes with relative links and other with absolute ones, MDI will only "standarize" any of them if it is moved or renamed.

So, there will be occasions, where a MDI branch will have both, absolute and relative, types of links.
I don't know yet how MDI should work in those cases.
if it should "normalize" all of them, depending on the baseFolder node attribute, each time it runs? or as a separated command?

Any ideas?

**Comment by bepolymathe** [comment]
Hi @EdoFro ,

Very good news. Thank you 👍🏼 

All this seems to me to be going in the right direction. 

> So, there will be occasions, where a MDI branch will have both, absolute and relative, types of links. I don't know yet how MDI should work in those cases. if it should "normalize" all of them, depending on the baseFolder node attribute, each time it runs? or as a separated command?

For this specific question I would tend to think that it would be simpler that the "normalization" takes place at each time it runs... for more automaticity. The only problem I can imagine would be in the case where the user has to make changes on both machines and the modification (sync by nextcloud for ex) is done while the "normalization" is taking place. In this case a dedicated command run manually is safer but more tedious. 🤔

**Reply by EdoFro** [reply]
I think I'm going to make a dedicated command first to see if it has colateral impacts.

thanks

**Reply by bepolymathe** [reply]
It is a wiser decision indeed. Available for testing.

**Reply by EdoFro** [reply]
v0.0.7 doesn't have the "normalization" command yet.

Hopefully in the next version.

**Reply by EdoFro** [reply]
added #23

**Comment by EdoFro** [comment]
Hi @bepolymathe,
I released a test version that should work with absolute and/or relative links. 
look here: #19
please test it with new maps first

**Reply by EdoFro** [reply]
new version #23

**Comment by EdoFro** [comment]
new version #25
